src/main/java/net/snowflake/ingest/streaming/internal/DataValidationUtil.java

```diff
@@ -802,6 +802,83 @@ static double validateAndParseReal(String columnName, Object input, long insertR
         columnName, input.getClass(), "REAL", new String[] {"Number", "String"}, insertRowIndex);
   }
 
+  /**
+   * Validates and parses input to a specific Iceberg column number type including INT, LONG, FLOAT,
+   * and DOUBLE. Allowed Java types:
+   *
+   * <ul>
+   *   <li>Number
+   *   <li>String
+   * </ul>
+   *
+   * @param columnName Column name, used in validation error messages
+   * @param input Object to validate and parse
+   * @param insertRowIndex Row index for error reporting
+   * @param targetType The type to which the input should be parsed ("INT", "LONG", "FLOAT",
+   *     "DOUBLE")
+   * @return Parsed number as Number to be casted by the caller
+   */
+  static <T extends Number> T validateAndParseIcebergNumber(
+      String columnName, Object input, long insertRowIndex, Class<T> targetType) {
```

Comment:
its easier to understand if you add different methods for the four types being passed in here. There is no commonality between the four classes being passed in here for them to share the same method other than the if (Number) else if (String) else throw structure to the method.

Reply:
Split into different methods.